src/frequenz/client/microgrid/component/_inverter.py

```diff
@@ -0,0 +1,153 @@
+        """Prevent instantiation of this class."""
+        if cls is Inverter:
+            raise TypeError(f"Cannot instantiate {cls.__name__} directly")
+        return super().__new__(cls)
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class UnspecifiedInverter(Inverter):
+    """An inverter of an unspecified type."""
+
+    type: Literal[InverterType.UNSPECIFIED] = InverterType.UNSPECIFIED
+    """The type of this inverter.
+
+    Note:
+        This should not be used normally, you should test if a inverter
+        [`isinstance`][] of a concrete inverter class instead.
+
+        It is only provided for using with a newer version of the API where the client
+        doesn't know about the new inverter type yet (i.e. for use with
+        [`UnrecognizedInverter`][frequenz.client.microgrid.component.UnrecognizedInverter]).
+    """
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class BatteryInverter(Inverter):
+    """A battery inverter."""
+
+    type: Literal[InverterType.BATTERY] = InverterType.BATTERY
+    """The type of this inverter.
+
+    Note:
+        This should not be used normally, you should test if a inverter
+        [`isinstance`][] of a concrete inverter class instead.
+
+        It is only provided for using with a newer version of the API where the client
+        doesn't know about the new inverter type yet (i.e. for use with
+        [`UnrecognizedInverter`][frequenz.client.microgrid.component.UnrecognizedInverter]).
+    """
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class SolarInverter(Inverter):
+    """A solar inverter."""
+
+    type: Literal[InverterType.SOLAR] = InverterType.SOLAR
+    """The type of this inverter.
+
+    Note:
+        This should not be used normally, you should test if a inverter
+        [`isinstance`][] of a concrete inverter class instead.
+
+        It is only provided for using with a newer version of the API where the client
+        doesn't know about the new inverter type yet (i.e. for use with
+        [`UnrecognizedInverter`][frequenz.client.microgrid.component.UnrecognizedInverter]).
+    """
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class HybridInverter(Inverter):
+    """A hybrid inverter."""
+
+    type: Literal[InverterType.HYBRID] = InverterType.HYBRID
+    """The type of this inverter.
+
+    Note:
+        This should not be used normally, you should test if a inverter
+        [`isinstance`][] of a concrete inverter class instead.
+
+        It is only provided for using with a newer version of the API where the client
+        doesn't know about the new inverter type yet (i.e. for use with
+        [`UnrecognizedInverter`][frequenz.client.microgrid.component.UnrecognizedInverter]).
+    """
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class UnrecognizedInverter(Inverter):
+    """An inverter component."""
+
+    type: int
+    """The unrecognized type of this inverter."""
+
+
+InverterTypes: TypeAlias = (
```

Comment:
Same

Reply:
Resolving this to keep the discussion in one thread, but whatever solution we agree on will be done for all `*Types` aliases.